### PR TITLE
Fix pagination by parsing the query parameters in gathering

### PR DESCRIPTION
**Pull request by cyli**
Rather than by appending the next link (which results in attempting to gather to URLs like:

http://localhost:8900/mimicking/NovaApi-92a5e6/ORD/v2/000000/http://localhost:8900/mimicking/NovaApi-92a5e6/ORD/v2/000000/servers/detail?marker=test-server1647753817-id-1647753817&limit=100

### CODE REVIEW

File: otter/test/convergence/test_gathering.py [hunk 1]
```diff
@@ -127,23 +137,25 @@ def test_get_all_with_link_to_next_page(self):
         svcreq = resolve_retry_stubs(get_all_server_details(batch_size=10))
         fake_response = object()
         body = {'servers': servers[:10],
-                'servers_links': [{'href': 'nexturl1', 'rel': 'next'}]}
-        result = resolve_svcreq(svcreq, (fake_response, body), *self.req)
+                'servers_links': [{'href': 'url?limit=10&marker=9',
```
Comment (radix) [inline otter/test/convergence/test_gathering.py]: I know it's not necessary, but I would feel better if this href had a full URL in it, like "https://example.com/url?..." to ensure the code is really ignoring everything in the URL before the query.

**Comment by radix** [comment]
Sorry I didn't catch this in my first review :( Other than the one nitpick LGTM.

**Comment by cyli** [comment]
@radix I totally forgot about `ServiceRequest` prepending stuff too. :(

**Comment by cyli** [comment]
jenkins please run integration tests

**Pull request merged**